packages/hyperion-react/src/IReactComponent.ts

```diff
@@ -0,0 +1,303 @@
+
+    this.name = component.displayName ?? component.name;
+  }
+}
+
+export const onReactClassComponentIntercept: Hook<
+  (shadow: ReactClassComponentShadowPrototype<IAny>) => void
+> = new Hook();
+
+export const onReactFunctionComponentIntercept: Hook<
+  <PropsType extends {}> (fi: FunctionInterceptor<ReactFunctionComponent<PropsType>, 'render', ReactFunctionComponent<PropsType>>) => void
+> = new Hook();
+
+export const onReactDOMElement: Hook<
+  (component: string, props: IAny) => void
+> = new Hook();
+
+export const onReactClassComponentElement: Hook<
+  (component: Class<ReactClassComponent<IAny>>, props: IAny) => void
+> = new Hook();
+
+export const onReactFunctionComponentElement: Hook<
+  (component: IReact.ReactStatelessFunctionalComponent<IAny>, props: IAny) => void
+> = new Hook();
+
+export const onReactSpecialObjectElement: Hook<
+  (component: ReactSpecialComponentTypes<IAny>, props: IAny) => void
+> = new Hook();
+
+export type InitOptions =
+  IReactElementVisitor.InitOptions &
+  Readonly<{
+    ReactModule: {
+      Component: typeof React.Component;
+    };
+    IReactModule: IReact.IReactModuleExports;
+    IJsxRuntimeModule: IReact.IJsxRuntimeModuleExports;
+  }>;
+
+const initialized = new TestAndSet();
+export function init(options: InitOptions): void {
+  if (initialized.testAndSet()) {
+    return;
+  }
+  const { ReactModule, IReactModule, IJsxRuntimeModule } = options;
+
+  IReactElementVisitor.init(options);
+  const interceptionInfo = new Map<
+    TReactClassComponent | Class<TReactClassComponent>,
+    ReactClassComponentShadowPrototype<PropsType> | null
+  >();
+
+  type PropsType = ReactComponentObjectProps;
+  type TReactClassComponent = ReactClassComponent<PropsType>;
+  type TReactFunctionComponent = IReact.ReactStatelessFunctionalComponent<PropsType>;
+
+  function processReactClassComponent(
+    component: Class<TReactClassComponent>,
+    classComponentParentClass: TReactClassComponent,
+  ): Class<TReactClassComponent> {
+    // For class components, we just need to intercept them once
+    if (interceptionInfo.has(component)) {
+      return component;
+    }
+    interceptionInfo.set(component, null);
+
+    if (!interceptionInfo.has(classComponentParentClass)) {
+      const info = new ReactClassComponentShadowPrototype(
+        component,
+        classComponentParentClass,
+      );
+      interceptionInfo.set(classComponentParentClass, info);
+      onReactClassComponentIntercept.call(info);
+    }
+    return component;
+  }
+
+  function processReactFunctionComponent(functionComponent: TReactFunctionComponent): TReactFunctionComponent {
+    /**
+     * For functional components, we should always replace them with the
+     * intercepted version of them.
+     * however, the interceptFunction itself will only assign a FunctionIntercetpr
+     * to the function once.
+     * So, we don't need to use the interceptionInfo map here.
+     */
+
+    const fi = __DEV__
```

Comment:
will `__DEV__` work as expected here?

Reply:
Yes, I made sure of that, it works both separately if you use the hyperion-test-react-app and also when we pull in the code into WWW